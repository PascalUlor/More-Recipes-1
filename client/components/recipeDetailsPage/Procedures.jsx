import React from 'react';
import PropTypes from 'prop-types';

const Procedures = (props) => {
  const procedures = props.procedures.split('.');
  return (
    <div className="col-md-7 col-lg-7">
      <h5 className="page-text font-weight-bold pb-2">Procedures</h5>
      <ul className="list-group list-group-flush text-muted details break-word">
        {procedures.map((procedure, index) => (
          (procedure.trim()) &&
          <li key={index}>
            <i className="fa fa-check-square-o pr-2 maroon" aria-hidden="true"></i>{procedure.trim()}
          </li>
        ))}
      </ul>
    </div>
  );
};

Procedures.propTypes = {
  procedures: PropTypes.string.isRequired
};

export default Procedures;
