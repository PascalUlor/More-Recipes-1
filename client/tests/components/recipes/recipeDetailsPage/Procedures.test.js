/* eslint-disable */
import React from 'react';
import Procedures
  from '../../../../components/recipeDetailsPage/Procedures.jsx';


const props = {
  procedures: 'boil water.'
};

describe('Procedures component', () => {
  it('should render correctly', () => {
    const mountWrapper = shallow(<Procedures {...props}/>);
    expect(mountWrapper).toMatchSnapshot();
  });
});
