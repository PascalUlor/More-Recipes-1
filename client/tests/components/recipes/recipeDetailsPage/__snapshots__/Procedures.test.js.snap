// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Procedures component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Procedures
    procedures="boil water."
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h5
          className="page-text font-weight-bold pb-2"
        >
          Procedures
        </h5>,
        <ul
          className="list-group list-group-flush text-muted details break-word"
        >
          <li>
            <i
              aria-hidden="true"
              className="fa fa-check-square-o pr-2 maroon"
            />
            boil water
          </li>
          
        </ul>,
      ],
      "className": "col-md-7 col-lg-7",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Procedures",
          "className": "page-text font-weight-bold pb-2",
        },
        "ref": null,
        "rendered": "Procedures",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <i
                aria-hidden="true"
                className="fa fa-check-square-o pr-2 maroon"
              />
              boil water
            </li>,
            "",
          ],
          "className": "list-group list-group-flush text-muted details break-word",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  aria-hidden="true"
                  className="fa fa-check-square-o pr-2 maroon"
                />,
                "boil water",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-check-square-o pr-2 maroon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "boil water",
            ],
            "type": "li",
          },
          "",
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="page-text font-weight-bold pb-2"
          >
            Procedures
          </h5>,
          <ul
            className="list-group list-group-flush text-muted details break-word"
          >
            <li>
              <i
                aria-hidden="true"
                className="fa fa-check-square-o pr-2 maroon"
              />
              boil water
            </li>
            
          </ul>,
        ],
        "className": "col-md-7 col-lg-7",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Procedures",
            "className": "page-text font-weight-bold pb-2",
          },
          "ref": null,
          "rendered": "Procedures",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <i
                  aria-hidden="true"
                  className="fa fa-check-square-o pr-2 maroon"
                />
                boil water
              </li>,
              "",
            ],
            "className": "list-group list-group-flush text-muted details break-word",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-check-square-o pr-2 maroon"
                  />,
                  "boil water",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-check-square-o pr-2 maroon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "boil water",
              ],
              "type": "li",
            },
            "",
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
